InstantOn standalone GA post to staging (#3154)
OpenLiberty/blogs: posts/2023-06-27-rapid-startup-instanton.adoc

-	The kernel must support the Linux link:https://man7.org/linux/man-pages/man7/capabilities.7.html[CAP_CHECKPOINT_RESTORE] capability. This capability was introduced in kernel version 5.9.
-	The latest available version of Podman for the Linux distribution must be installed.
-	The Linux distribution must allow the execution of privileged container builds by using Podman or Docker.

For more information about the runtime and host build system prerequisites, see the link:/docs/latest/instanton.html#prereq[Open Liberty InstantOn documentation].

== Create an application WAR file

If you don’t have an application of your own handy, you can follow along with an example application from the link:/guides/getting-started.html[Getting started with Open Liberty guide].

First, clone the link:https://github.com/openliberty/guide-getting-started[Git repository] for the guide:

[source,console]
----
git clone https://github.com/openliberty/guide-getting-started.git
cd guide-getting-started
----

Then, build the application, which is in the `finish/` directory, and deploy it to Open Liberty:

[source,console]
----
cd finish
mvn liberty:run
----

> When you see the following message, your application server is ready:

Reviewer: ```suggestion
When you see the following message, your Open Liberty instance is ready:
```

[source,console]
----
The defaultServer server is ready to run a smarter planet.
----

Check out the service at the http://localhost:9080/dev/system/properties URL.

> Stop the Open Liberty server by pressing **CTRL+C** in the command-line session where you ran the server.

Reviewer: ```suggestion
Stop the Open Liberty server by pressing **CTRL+C** in the command-line session where you ran the server.
```
```suggestion
Stop the running Open Liberty instance by pressing **CTRL+C** in the command-line session where you started Open Liberty.
```

== Testing the startup time of your application

For comparison of how long it takes your Open Liberty application container image to start both with and without InstantOn, we describe how to build the container image without InstantOn first. Then, we describe how to build with InstantOn and run the resulting containers.

=== Containerizing the Open Liberty application without InstantOn

Build the application container image without InstantOn:

[source,console]
----
podman build -t getting-started .
----

This command creates the getting-started container image without any checkpoint image. 

Run this application container:

[source,console]
----
podman run --name getting-started --rm -p 9080:9080 getting-started
----

Note the amount of time Open Liberty takes to report it is started and check out the service running in the container at the http://localhost:9080/dev/system/properties URL. After you finish checking out the application, stop the running container by pressing **CTRL+C** in the command-line session where you ran the `podman run` command.

=== Containerizing the Open Liberty application with InstantOn

> The Open Liberty container image contains the prerequisites for building an application container image with a checkpointed server process. Applications can use the Open Liberty image as a base to build their own application container images and from that, create their own application container image with a checkpointed process. 

Reviewer: ```suggestion
The Open Liberty container image contains the prerequisites for building an application container image with a checkpointed runtime process. Applications can use the Open Liberty image as a base to build their own application container images and from that, create their own application container image with a checkpointed process. 
```
